src/components/com_tjcertificate/site/views/certificates/tmpl/my.php

```diff
@@ -85,6 +84,21 @@
 
 							foreach ($this->items as $i => $item)
 							{
+								$name = '';
+
+								if ($item->client === "com_tjlms.course")
+								{
+									JLoader::import('components.com_tjlms.models.orders', JPATH_SITE);
+									$ordersModel = JModelLegacy::getInstance('Orders', 'TjlmsModel', array('ignore_request' => true));
+									$name = $ordersModel->getCourseName($item->client_id);
+								}
+								elseif ($item->client === "com_jticketing.event")
+								{
+									JLoader::import('components.com_jticketing.helpers.event', JPATH_SITE);
+									$JteventHelper = new JteventHelper;
+									$name = $JteventHelper->getEventData($item->client_id)->title;
```

Comment:
I think here also you can use the same plugin function `getCertificateClientData` to fetch the client details...In this way, any third party extension won't have to hack it

Reply:
done